### PR TITLE
Editorial: Move AsyncFunctionCreate into 9.2 ECMAScript Function

### LINKED ISSUE #1171: Move AsyncFunctionCreate into 9.2 ECMAScript Function Objects
[25.7.5.1 AsyncFunctionCreate](https://tc39.github.io/ecma262/#sec-async-functions-abstract-operations-async-function-create) should probably be moved into [9.2 ECMAScript Function Objects](https://tc39.github.io/ecma262/#sec-ecmascript-function-objects) for consistency with FunctionCreate, GeneratorFunctionCreate, and AsyncGeneratorFunctionCreate.

### PR DESCRIPTION
**Pull request by akoserwal**
Fixes #1171

### CODE REVIEW

**Comment by akoserwal** [comment]
#1171

File: spec.html [hunk 1]
```diff
@@ -7679,7 +7691,7 @@ <h1>AsyncGeneratorFunctionCreate (_kind_, _ParameterList_, _Body_, _Scope_, _Str
       </emu-alg>
     </emu-clause>
 
-    <!-- es6num="9.2.7" -->
+    <!-- es6num="9.2.8" -->
```
Comment (ljharb) [inline spec.html]: Please revert all of these es6num changes; these are legacy links from ES6 and should never be renumbered.

Comment (ljharb) [reply spec.html]: No, things that weren’t in es6 shouldnt have es6nums

**Comment by ljharb** [comment]
cc @anba

**Comment by anba** [comment]
:+1: Thanks!

**Comment by bterlson** [comment]
Looks good, thanks!

@akoserwal congrats on your first ECMA262 contribution!! 🎉🎉

**Comment by akoserwal** [comment]
@bterlson: thanks, looking forward to keep up the contribution :smile:

**Pull request merged**